Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -249,6 +249,21 @@ public async ValueTask<bool> ExistsAsync(Expression<Func<TItem, bool>> predicate
                 item => item.Type == typeof(TItem).Name, Expression.AndAlso).Compile());
         }
 
+        /// <inheritdoc/>
+        public async ValueTask<int> CountAsync(CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            return Items.Values.Count;
```

Comment:
Should this be checking the type of the items count, say you have multiple `TItem`s within the same container? Depends on the intention of the API I think.

Reply:
You're correct. I will make the update. In the `DefaultRepository` it filters by type.

Reply:
I've added the filter in the implementation.